src/network-analytics/azext_network_analytics/aaz/latest/network_analytics/data_product/_ingest.py

```diff
@@ -0,0 +1,168 @@
+        cls._args_schema = super()._build_arguments_schema(*args, **kwargs)
+
+        # define Arg Group ""
+
+        _args_schema = cls._args_schema
+        _args_schema.data_product_name = AAZStrArg(
+            options=["--data-product-name"],
+            help="The data product resource name",
+            required=True,
+            id_part="name",
+            fmt=AAZStrArgFormat(
+                pattern="^[a-z][a-z0-9]*$",
+                max_length=63,
+                min_length=3,
+            ),
+        )
+        _args_schema.resource_group = AAZResourceGroupNameArg(
+            required=True,
+        )
+
+        # define Arg Group "Body"
+
+        _args_schema = cls._args_schema
+        _args_schema.data_type = AAZStrArg(
+            options=["--data-type"],
+            arg_group="Body",
+            help="Data Type.",
+            required=True,
+        )
+        _args_schema.source = AAZStrArg(
+            options=["--srcdir"],
+            arg_group="Body",
+            help="Source directory path.",
+            required=True,
+        )
+        return cls._args_schema
+    
+    def _execute_operations(self):
+        self.pre_operations()
+        self.DataProductsIngest(ctx=self.ctx)()
+        self.post_operations()
+
+    @register_callback
+    def pre_operations(self):
+        pass
+
+    @register_callback
+    def post_operations(self):
+        pass
+
+    def _output(self, *args, **kwargs):
+        result = self.deserialize_output(self.ctx.vars.instance, client_flatten=True)
+        return result
+    
+    class DataProductsIngest(object):
+        DATA_PRODUCT_ARM_ID = "/subscriptions/{}/resourceGroups/{}/providers/Microsoft.NetworkAnalytics/dataProducts/{}"
+        API_VERSION = "2023-11-15"
+        KEYVAULT_NAME = "input-storage-sas"
+        KEYVAULT_URI = "https://aoi-{}-kv.vault.azure.net/"
+
+        def __init__(self, ctx):
+            self.ctx = ctx
+            self.subscription_id = ctx.subscription_id
+            self.resource_group = ctx.args.resource_group
+            self.data_product_name = ctx.args.data_product_name
+            self.data_type = ctx.args.data_type
+            self.source = ctx.args.source
+
+            self.resources_client = get_mgmt_service_client(self.ctx.cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES, subscription_id=self.subscription_id).resources
+            self.container_client = get_sdk(self.ctx.cli_ctx, ResourceType.DATA_STORAGE_BLOB, '_container_client#ContainerClient')
+
+        def __call__(self, *args, **kwargs):
+            data_product = self.get_data_product()
+            secret = self.get_key_vault_secret(data_product)
+            self.upload_file(secret)
+
+        def get_data_product(self):
+            arm_id = self.DATA_PRODUCT_ARM_ID.format(self.subscription_id, self.resource_group, self.data_product_name)
+            resource = self.resources_client.get_by_id(arm_id, self.API_VERSION)
+            return resource
```

Comment:
I understand that it makes the code cleaner, but I am not sure if it might add clutter to the overall code. I am open for discussing this idea though. I would like to know your thoughts behind it.